engine/src/main/java/org/hibernate/validator/internal/metadata/core/MetaConstraint.java

```diff
@@ -41,13 +45,28 @@
 	private final ConstraintLocation location;
 
 	/**
+	 * The {@link ValueExtractor} used to extract the value for validation.
+	 */
+	private final ValueExtractorDescriptor valueExtractorDescriptor;
+
+	/**
+	 * The type of the validated element.
+	 */
+	private final Type typeOfValidatedElement;
```

Comment:
How is it different from `ConstraintLocation#getTypeForValidatorResolution()`? Having all these types with slightly different semantics makes me a bit concerned.